### PR TITLE
[N/A] Hook up event propagation for all browserview-related events

### PR DESCRIPTION
**Pull request by MichaelMCoates**
#### Description of Change

This PR adds propagated events up to Application and System for BrowserView.

Tests:
https://testing-dashboard.openfin.co/#/app/sessions/api/completed/5da8b1ceee166d7a14d451bc

Corresponding core PR:
https://github.com/HadoukenIO/core/pull/976

@aziz512 trying to add you as reviewer but it won't let me for some reason.

#### Checklist

- [X] PR description included and stakeholders cc'd
- [X] `npm test` passes
- [X] automated tests are [changed or added](https://testing-dashboard.openfin.co/#/app/dashboard)
- [X] relevant documentation is [changed or added](https://github.com/hadoukenio/js-adapter)
- [X] Link to new tests added
- [X] PR has assigned reviewers

#### Release Notes

Notes:

### CODE REVIEW

**Review by dhamberlin — DISMISSED** [review]

File: src/api/events/window.ts [hunk 1]
```diff
@@ -200,14 +199,27 @@ export interface PropagatedWindowEventMapping<Topic = string, Type = string> ext
...
+    'window-view-attached': WindowEvent<Topic, Type>;
+    'window-view-crashed': CrashedEvent & WindowEvent<Topic, Type>;
+    'window-view-created': WindowEvent<Topic, Type>;
+    'window-view-destroyed': WindowEvent<Topic, Type>;
+    'window-view-detached': WindowEvent<Topic, Type>;
+    'window-view-did-change-theme-color': WindowEvent<Topic, Type>;
+    'window-view-hidden': WindowEvent<Topic, Type>;
+    'window-view-page-favicon-updated': WindowEvent<Topic, Type>;
+    'window-view-page-title-updated': WindowEvent<Topic, Type>;
+    'view-resource-load-failed': WindowResourceLoadFailedEvent<Topic, Type>;
```
Comment (licui3936) [inline src/api/events/window.ts]: 'window' is missing in the event name?

File: src/api/events/window.ts [hunk 2]
```diff
@@ -200,14 +199,27 @@ export interface PropagatedWindowEventMapping<Topic = string, Type = string> ext
...
+    'view-resource-response-received': WindowResourceResponseReceivedEvent<Topic, Type>;
```
Comment (licui3936) [inline src/api/events/window.ts]: same

File: tutorials/Application.EventEmitter.md [hunk 3]
```diff
@@ -110,23 +111,46 @@ app.removeAllListeners("closed");
...
+* window-resource-load-failed (see {@tutorial Window.EventEmitter})
+* window-resource-response-received (see {@tutorial Window.EventEmitter})
 * window-responding
 * window-restored (see {@tutorial Window.EventEmitter})
 * window-show-requested
 * window-shown (see {@tutorial Window.EventEmitter})
 * window-start-load
 * window-user-movement-disabled (see {@tutorial Window.EventEmitter})
 * window-user-movement-enabled (see {@tutorial Window.EventEmitter})
+* window-view-attached (see {@tutorial Window.EventEmitter})
```
Comment (licui3936) [inline tutorials/Application.EventEmitter.md]: @tutorial Window.EventEmitter => @tutorial BrowserView.EventEmitter

File: tutorials/Application.EventEmitter.md [hunk 4]
```diff
@@ -110,23 +111,46 @@ app.removeAllListeners("closed");
...
+* window-view-crashed (see {@tutorial BrowserView.EventEmitter})
+* window-view-created (see {@tutorial BrowserView.EventEmitter})
+* window-view-destroyed (see {@tutorial BrowserView.EventEmitter})
+* window-view-detached (see {@tutorial Window.EventEmitter})
```
Comment (licui3936) [inline tutorials/Application.EventEmitter.md]: same here

Comment (MichaelMCoates) [reply tutorials/Application.EventEmitter.md]: All fixed. Thank you!

**Comment by MichaelMCoates** [comment]
After talking with Pierre and Tommy, changed the eventing scheme to the following:

- Windows can listen to `view-attached` and `view-detached`.
- Views can listen to `target-changed`. 
- Window `view` events don't propagate up to Application and System.
- All `view` events propagate up to Window, Application, and System.

The tests have been updated to reflect these changes.

**Pull request closed without merging**